lib/AST/ASTPrinter.cpp

```diff
@@ -72,19 +72,39 @@ PrintOptions PrintOptions::printTextualInterfaceFile() {
   result.FullyQualifiedTypes = true;
   result.SkipImports = true;
 
-  class UsableFromInlineOnly : public ShouldPrintChecker {
-    bool shouldPrint(const Decl *D, PrintOptions &options) override {
+  class ShouldPrintForTextualInterface : public ShouldPrintChecker {
+    bool shouldPrint(const Decl *D, const PrintOptions &options) override {
+      // Skip anything that isn't 'public' or '@usableFromInline'.
       if (auto *VD = dyn_cast<ValueDecl>(D)) {
         AccessScope accessScope =
             VD->getFormalAccessScope(/*useDC*/nullptr,
                                      /*treatUsableFromInlineAsPublic*/true);
         if (!accessScope.isPublic())
           return false;
       }
+
+      // Skip typealiases that just redeclare generic parameters.
```

Comment:
Oh, gross. I assume these are getting created by associated type inference, which is gross. I went ahead and filed https://bugs.swift.org/browse/SR-8564 to stop doing this, but for now your hack is reasonable to make progress.